src/pages/workspaces/workspace/WorkspaceMenu.ts

```diff
@@ -6,12 +6,18 @@ import { Clickable } from 'src/components/common';
 import { MenuButton } from 'src/components/MenuButton';
 import { makeMenuIcon, MenuTrigger } from 'src/components/PopupTrigger';
 import { useWorkspaceDetails } from 'src/components/workspace-utils';
-import { isOwner, WorkspacePolicy, WorkspaceWrapper as Workspace } from 'src/libs/workspace-utils';
+import { isOwner, WorkspacePolicy, WorkspaceState, WorkspaceWrapper as Workspace } from 'src/libs/workspace-utils';
 
 const isNameType = (o: WorkspaceInfo): o is DynamicWorkspaceInfo =>
   'name' in o && typeof o.name === 'string' && 'namespace' in o && typeof o.namespace === 'string';
 
-type LoadedWorkspaceInfo = { canShare: boolean; isLocked: boolean; isOwner: boolean; workspaceLoaded: boolean };
+type LoadedWorkspaceInfo = {
+  state?: WorkspaceState;
```

Comment:
Because of how useWorkspaceDetails works, it isn't guaranteed.

There's a window of time between the component being rendered and workspace details being loaded in which we might not have it.

This isn't as obvious for the other properties, because they're all booleans.